osv/purl_helpers.py

```diff
@@ -63,7 +107,8 @@ def package_to_purl(ecosystem: str, package_name: str) -> str | None:
   return f'pkg:{purl_type}/{_url_encode(package_name)}{suffix}'
 
 
-def purl_to_ecosystem(purl_type: str) -> str | None:
-  """Convert purl to a specific ecosystem string"""
-  ecosystem_purl = {v: k for k, v in PURL_ECOSYSTEMS.items()}
-  return ecosystem_purl.get(purl_type)
+def parse_purl_ecosystem(purl: PackageURL) -> str | None:
+  """Extracts the ecosystem name from a PackageURL by checking
+  its `type` and `namespace`."""
+  return PURL_TYPE_ECOSYSTEMS.get(
+      purl.type, PURL_NAMESPACE_ECOSYSTEMS.get(purl.namespace, None))
```

Comment:
fwiw, I think this will convert invalid purls e.g. `pkg:badtype/ubuntu/curl` -> `Ubuntu`

This function actually seems like a good candidate to use Python's new-ish [`match` statements](https://docs.python.org/3/tutorial/controlflow.html#match-statements):
```python
def parse_purl_ecosystem(purl: PackageURL) -> str | None:
  match purl:
    case PackageURL(type='cargo'):
      return 'crates.io'
    case PackageURL(type='deb', namespace='debian'):
      return 'Debian'
    case PackageURL(type='deb', namespace='ubuntu'):
      return 'Ubuntu'
    case _:
      return None
```

You can also do something like this to grab fields into variables:
```python
match purl:
  case PackageURL(type='cargo', name=pkg, version=ver):
    return 'crates.io', pkg, ver
```

Reply:
updated, thanks!